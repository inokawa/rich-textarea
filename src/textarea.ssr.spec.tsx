/**
 * @jest-environment node
 */
import { it, describe, expect } from "vitest";
import { renderToString, renderToStaticMarkup } from "react-dom/server";
import { RichTextarea } from ".";

const NOP = () => {};

describe("SSR", () => {
  it("should succeed with renderToString", () => {
    expect(
      renderToString(
        <RichTextarea
          value={"Lorem ipsum dolor sit amet"}
          style={{ width: "123px", height: "456px" }}
          onChange={NOP}
        >
          {(v) => v}
        </RichTextarea>
      )
    ).toMatchSnapshot();
  });

  it("should succeed with renderToStaticMarkup", () => {
    expect(
      renderToStaticMarkup(
        <RichTextarea
          value={"Lorem ipsum dolor sit amet"}
          style={{ width: "123px", height: "456px" }}
          onChange={NOP}
        >
          {(v) => v}
        </RichTextarea>
      )
    ).toMatchSnapshot();
  });
});
