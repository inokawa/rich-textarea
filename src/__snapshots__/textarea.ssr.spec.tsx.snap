// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SSR should succeed with renderToStaticMarkup 1`] = `"<div style="display:inline-block;position:relative;width:0;height:0"><div style="position:absolute;overflow:hidden;top:0;left:0;width:0;height:0"><div aria-hidden="true" style="width:0;transform:translate(0px, 0px);pointer-events:none;user-select:none;-ms-user-select:none;-webkit-user-select:none;box-sizing:content-box;text-size-adjust:100%;-webkit-text-size-adjust:100%">Lorem ipsum dolor sit amet<span style="color:transparent">​</span></div></div><textarea style="width:123px;height:456px;background:transparent;margin:0;vertical-align:top">Lorem ipsum dolor sit amet</textarea></div>"`;

exports[`SSR should succeed with renderToString 1`] = `"<div style="display:inline-block;position:relative;width:0;height:0"><div style="position:absolute;overflow:hidden;top:0;left:0;width:0;height:0"><div aria-hidden="true" style="width:0;transform:translate(0px, 0px);pointer-events:none;user-select:none;-ms-user-select:none;-webkit-user-select:none;box-sizing:content-box;text-size-adjust:100%;-webkit-text-size-adjust:100%">Lorem ipsum dolor sit amet<span style="color:transparent">​</span></div></div><textarea style="width:123px;height:456px;background:transparent;margin:0;vertical-align:top">Lorem ipsum dolor sit amet</textarea></div>"`;
